File: 2657-find-the-prefix-common-array-of-two-arrays/2657-find-the-prefix-common-array-of-two-arrays.py
class Solution:
    def findThePrefixCommonArray(self, A: List[int], B: List[int]) -> List[int]:
        ans = []
        n = len(A)
        setA, setB = set(), set()
        cnt = 0
        for i in range(n):
            setA.add(A[i])
            setB.add(B[i])
            if A[i] in setB:
                cnt += 1
            if B[i] in setA:
                cnt += 1
            if A[i] == B[i]:
                cnt -= 1
            ans.append(cnt)
            
        return ans

File: 2657-find-the-prefix-common-array-of-two-arrays/2657-find-the-prefix-common-array-of-two-arrays.cpp
class Solution {
public:
    vector<int> findThePrefixCommonArray(vector<int>& A, vector<int>& B) {
        vector<int> ans;

        int n = A.size();
        int cnt = 0;
        unordered_set<int> setA, setB;

        for(int i=0; i<n; i++){
            setA.insert(A[i]);
            setB.insert(B[i]);
            if(setA.find(B[i]) != setA.end()) cnt++;
            if(setB.find(A[i]) != setB.end()) cnt++;
            if(A[i] == B[i]) cnt--;
            ans.push_back(cnt);
        }
        return ans;
    }
};